File: exercism_data/src/510.rb
class Phrase
  CHAR_MASK = /[^[a-z][0-9]']/

  def initialize(string)
    @words = string.downcase.gsub(CHAR_MASK, ' ').split
  end

  def word_count
    Hash.new(0).tap do |counts|
      @words.each { |word| counts[word] += 1 }
    end
  end
end

File: assignments/python/wc/src/510.py
def word_count(prompt):
    wordArray = prompt.split()
    wordCount = {}
    for word in wordArray:
        currentValue = 0
        try:
            currentValue = wordCount[word]
        except:
            pass
        wordCount.update({word: currentValue + 1})
    return wordCount
